src/librustc/hir/map/definitions.rs

```diff
@@ -466,7 +466,11 @@ impl Definitions {
     }
 
     pub fn find_node_for_hir_id(&self, hir_id: hir::HirId) -> ast::NodeId {
```

Comment:
Can't you rather do some sort of merging of this function with `validate_hir_id`? Unmarked linear searches are dangerous.